### PR TITLE
Orca: fix np 1.24 type deprecations

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

See https://github.com/intel-analytics/BigDL/issues/7548

### 3. Summary of the change 
 
fix numpy deprecations:
+ `np.int` -> `np.int32`
+ `np.float` -> `np.float32`
+ `np.str` -> `str`

### 4. How to test?

The following Orca unit test:

- [ ] Orca-Pytest-Ray-Py38-Spark3
- [ ] Orca-Pytest-Tf1-Py38-Spark3

### CODE REVIEW

File: python/orca/dev/test/run-pytests-ray.sh [hunk 1]
```diff
@@ -45,7 +45,8 @@ python -m pytest -v test/bigdl/orca/learn/ray \
       --ignore=test/bigdl/orca/learn/ray/pytorch/test_estimator_horovod_backend.py \
       --ignore=test/bigdl/orca/learn/ray/pytorch/test_estimator_ray_runtime.py \
       --ignore=test/bigdl/orca/learn/ray/pytorch/test_estimator_ray_dataset.py \
-      --ignore=test/bigdl/orca/learn/ray/tf/
+      --ignore=test/bigdl/orca/learn/ray/tf/ \
+      --ignore=test/bigdl/orca/learn/ray/mxnet/
```
Comment (hamham223) [inline python/orca/dev/test/run-pytests-ray.sh:49]: WIP: need to add mxnet elsewhere

File: python/orca/src/bigdl/orca/learn/utils.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def group_index(iter):
     def transform_predict(predictions):
         # list of np array
         if isinstance(predictions[0], list):
-            predictions = np.array(predictions).T.tolist()
+            predictions = np.array(predictions, dtype=object).T.tolist()
```
Comment (hamham223) [inline python/orca/src/bigdl/orca/learn/utils.py:97]: np 1.24 requires `array` to have homogenous shape

File: python/orca/dev/test/run-pytests-ray.sh (hunk 1 above)
Comment (hkvision) [reply python/orca/dev/test/run-pytests-ray.sh:49]: Refer to other jobs to add if condition for py37; we can only run mxnet for py37

File: python/orca/src/bigdl/orca/learn/openvino/estimator.py [hunk 3]
```diff
@@ -142,10 +142,10 @@ def generate_output_row(batch_input_dict):
                     elif isinstance(feature_type, df_types.IntegerType):
                         input_dict[input], elem_num = add_elem(np.array(value).astype(np.int32))
                     elif isinstance(feature_type, df_types.StringType):
-                        input_dict[input], elem_num = add_elem(np.array(value).astype(np.str))
+                        input_dict[input], elem_num = add_elem(np.array(value).astype(str))
                     elif isinstance(feature_type, df_types.ArrayType):
                         if isinstance(feature_type.elementType, df_types.StringType):
-                            input_dict[input], elem_num = add_elem(np.array(value).astype(np.str))
+                            input_dict[input], elem_num = add_elem(np.array(value).astype(str))
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/openvino/estimator.py:148]: @cyita take a look?

File: python/orca/src/bigdl/orca/learn/utils.py (hunk 2 above)
Comment (hkvision) [reply python/orca/src/bigdl/orca/learn/utils.py:97]: not correct to be here; we can't treat everything to be object.

File: python/orca/src/bigdl/orca/learn/openvino/estimator.py (hunk 3 above)
Comment (cyita) [reply python/orca/src/bigdl/orca/learn/openvino/estimator.py:148]: ov est LGTM

File: python/orca/src/bigdl/orca/learn/utils.py (hunk 2 above)
Comment (hamham223) [reply python/orca/src/bigdl/orca/learn/utils.py:97]: python 3.7 test already gives the warning:
```python
/home/yinchen/BigDL/python/orca/src/bigdl/orca/learn/utils.py:97: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  predictions = np.array(predictions).T.tolist()
```
Command
```bash
cd python/orca
python -m pytest --show-capture=all --full-trace -r P -v test/bigdl/orca/learn/test_utils.py::TestUtil::test_convert_predict_rdd_to_xshard_multi_output
```

Comment (hkvision) [reply python/orca/src/bigdl/orca/learn/utils.py:97]: @sgwhat Fix this issue?
```
predictions = np.array(predictions).T.tolist()
```

File: python/orca/src/bigdl/orca/learn/utils.py [hunk 4]
```diff
@@ -94,7 +94,7 @@ def group_index(iter):
     def transform_predict(predictions):
         # list of np array
         if isinstance(predictions[0], list):
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/utils.py:96]: fixed here. @sgwhat take a look.

File: python/orca/src/bigdl/orca/learn/utils.py [hunk 5]
```diff
@@ -94,7 +94,7 @@ def group_index(iter):
...
-            predictions = np.array(predictions).T.tolist()
+            predictions = [[i[j] for i in predictions] for j in range(len(predictions[0]))]
             result = [np.array(predict) for predict in predictions]
             return result
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/utils.py:99]: rename `j` to `i` and rename `i` to `item`, this will be clearer?

Comment (hkvision) [reply python/orca/src/bigdl/orca/learn/utils.py:99]: @sgwhat Fixed here, take a look.

Comment (hkvision) [reply python/orca/src/bigdl/orca/learn/utils.py]: @cyita Take a look?

File: python/orca/src/bigdl/orca/learn/utils.py [hunk 6]
```diff
@@ -94,7 +94,7 @@ def group_index(iter):
     def transform_predict(predictions):
         # list of np array
         if isinstance(predictions[0], list):
-            predictions = np.array(predictions).T.tolist()
+            predictions = [[item[i] for item in predictions] for i in range(len(predictions[0]))]
```
Comment (sgwhat) [inline python/orca/src/bigdl/orca/learn/utils.py:97]: will predictions return as shape `[[1, 2, 3], [4, 5, 6], [7, 8]]` ?

Comment (hamham223) [reply python/orca/src/bigdl/orca/learn/utils.py:97]: Before running Line 97:
`predictions = [[[1, 1], [1, 1, 1]], [[2, 2], [2, 2, 2]]]`
After Line 97:
`predictions = [[[1, 1], [2, 2]], [[1, 1, 1], [2, 2, 2]]]`

**Review by hkvision — APPROVED** [review]

**Comment by hkvision** [comment]
Merge this first, one ray ctx job is irrelevant.

**Pull request merged**